app/utils/Execution.scala

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils
+
+import java.util
+
+import scala.concurrent.{ExecutionContext, ExecutionContextExecutor}
+
+object Execution {
+
+  def defaultExecutionContext: ExecutionContext = Implicits.defaultExecutionContext
+
+  object Implicits {
+    implicit def defaultExecutionContext: ExecutionContext = Execution.Trampoline
+
+    implicit def trampoline: ExecutionContextExecutor = Execution.Trampoline
+  }
+  object Trampoline extends ExecutionContextExecutor {
+    private val local = new ThreadLocal[AnyRef]
+    private object Empty
+    private def executeScheduled(): Unit = local.get match {
```

Comment:
You can add the `tailrec` annotation to optimise this (accepting IntelliJ hint will do this automatically)